packages/web3-eth-accounts/src/index.js

```diff
@@ -168,36 +154,28 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
             transaction.value = transaction.value || '0x';
             transaction.chainId = utils.numberToHex(transaction.chainId);
 
-            var rlpEncoded = RLP.encode([
-                Bytes.fromNat(transaction.nonce),
-                Bytes.fromNat(transaction.gasPrice),
-                Bytes.fromNat(transaction.gas),
-                transaction.to.toLowerCase(),
-                Bytes.fromNat(transaction.value),
-                transaction.data,
-                Bytes.fromNat(transaction.chainId || "0x1"),
-                "0x",
-                "0x"]);
-
-
-            var hash = Hash.keccak256(rlpEncoded);
+            if (privateKey.startsWith('0x')) {
+                privateKey = privateKey.substring(2);
+            }
 
-            var signature = Account.makeSigner(Nat.toNumber(transaction.chainId || "0x1") * 2 + 35)(Hash.keccak256(rlpEncoded), privateKey);
+            var ethTx = new Transaction(transaction);
+            ethTx.sign(Buffer.from(privateKey, 'hex'));
 
-            var rawTx = RLP.decode(rlpEncoded).slice(0, 6).concat(Account.decodeSignature(signature));
+            var validationResult = ethTx.validate(true);
 
-            rawTx[6] = makeEven(trimLeadingZero(rawTx[6]));
-            rawTx[7] = makeEven(trimLeadingZero(rawTx[7]));
-            rawTx[8] = makeEven(trimLeadingZero(rawTx[8]));
+            if (validationResult !== '') {
+                throw new Error('Signer Error: ' + validationResult);
+            }
```

Comment:
What do you think about adding the example tx in #1074 to the unit tests to see if it hits this validation check? 

It looks like this:
```js
signTxParams = { 
    data: "0x23123123123123123",  
    gas: 42424242424242,
    gasPrice: 42424242,
    nonce: 1,
    to: "0xd8b0990c007ba1ad97b37c001d1f87044312162a",
    value: "42424242",
    chainId: 1
}
```